### PR TITLE
27 - Getting an order by id

### PR DESCRIPTION
**Pull request by aMotherFox**

### CODE REVIEW

File: java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/entree/EntreeRepository.java [hunk 1]
```diff
@@ -89,4 +89,16 @@ public List<Entree> findAllByIds(List<Integer> entreeIds) {
...
         return namedParameterJdbcTemplate.query(sql, parameters, new BeanPropertyRowMapper<>(Entree.class));
     }
+
+    public List<Entree> findEntreeThroughEntreeOrdered(Integer id) {
+
+        String sql = "SELECT entree.* " +
+                " FROM entree" +
+                " JOIN entree_ordered ON entree.id = entree_ordered.entree_id" +
+                " JOIN \"order\" ON \"order\".id = entree_ordered.order_id" +
+                " WHERE \"order\".id = ?";
```
Comment (LucasCostaAtCyberSaint) [inline java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/entree/EntreeRepository.java]: I believe this is ok, but I think you are unnecessarily joining the "order" table. You already know the "order".id, from the `entree_ordered` table. Can you just find your entrees from that instead?

File: java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/order/OrderService.java [hunk 2]
```diff
@@ -40,6 +40,23 @@ public OrderDTO createOrder(NewOrderRequestBody newOrderRequestBody) {
...
         );
     }
+
+    public OrderDTO getOrderById(Integer id) {
+        Order orderById = orderRepository.getOrderById(id);
+
+        List<Entree> entrees = entreeRepository.findEntreeThroughEntreeOrdered(id);
+        System.out.println("Entrees ORDER SERVICE: " + entrees);
+        List<Appetizer> appetizers = appetizerRepository.findAppetizerThroughAppetizerOrdered(id);
+        System.out.println("Apps ORDER SERVICE: " + appetizers);
```
Comment (LucasCostaAtCyberSaint) [inline java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/order/OrderService.java]: would we prefer some logs here instead?

File: java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/appetizer/AppetizerRepository.java [hunk 3]
```diff
@@ -89,4 +89,15 @@ public List<Appetizer> findAllByIds(List<Integer> appetizerIds) {
...
         return namedParameterJdbcTemplate.query(sql, parameters, new BeanPropertyRowMapper<>(Appetizer.class));
     }
+
+    public List<Appetizer> findAppetizerThroughAppetizerOrdered(Integer id) {
+
+        String sql = "SELECT appetizer.* " +
+                " FROM appetizer" +
+                " JOIN appetizer_ordered ON appetizer.id = appetizer_ordered.appetizer_id" +
+                " JOIN \"order\" ON \"order\".id = appetizer_ordered.order_id" +
+                " WHERE \"order\".id = ?";
```
Comment (LucasCostaAtCyberSaint) [inline java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/appetizer/AppetizerRepository.java]: same thing here, I think we don't need to join the order table

File: java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/order/OrderService.java (hunk 2 above)
Comment (aMotherFox) [reply java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/order/OrderService.java]: Yes, they have been updated to logs

File: java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/appetizer/AppetizerRepository.java (hunk 3 above)
Comment (aMotherFox) [reply java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/appetizer/AppetizerRepository.java]: this is been condensed, thanks

File: java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/entree/EntreeRepository.java (hunk 1 above)
Comment (aMotherFox) [reply java/FoodTruckApp/src/main/java/com/my/FoodTruckApp/entree/EntreeRepository.java]: this has been condensed, thank you

**Review by LucasCostaAtCyberSaint — APPROVED** [review]

**Pull request merged**